### ISSUE TITLE
Create a "saved filter" custom field

### ISSUE THREAD

**Issue by goraxan** [masugadesign/cp-filters-craft-cms#18]
I'm trying to extend the saved filters with custom fields. One approach is to create a custom field extending the craft\fields\BaseRelationField class (similar to how Assets, Categories, Entries, Tags and Users fields do) so that you can add a saved filter as a field of a section along with the rest of the fields I need (in my case an image to show a "preview" to the user, and other fields).

I've managed to create the field after doing the following modifications to the Masuga\CpFilters\elements\SavedFilter element:

This methods need to be defined in order to show the results in a table
```
	protected static function defineTableAttributes(): array
	{
		return [
			'title' => \Craft::t('app', 'Title'),
			'filterElementType' => \Craft::t('cpfilters', 'Type')
		];
	}
```

This method need to return an array
```
	protected static function defineSources(string $context = null): array
	{
		$sources = [
			[
				'key'      => '*',
				'label'    => Craft::t('cpfilters', 'Saved Filters'),
				'defaultSort' => ['elements.dateCreated', 'desc']
			],
		];

		return $sources;
	}
```

There was a typo in the i18n key
```
	protected static function defineSortOptions(): array
	{
		return [
			'elements.dateCreated' => Craft::t('app', 'Date Created'),
			'title' => Craft::t('cpfilters', 'Title')
		];
	}
```

I'm happy to create a PR if you find it useful.

After doing these changes I could create the custom field and use it in my entries without any problem. This is an example of a CP filter called "Artwork" being used in one of my entries
![image](https://user-images.githubusercontent.com/8337784/145044291-8fc13b1f-2860-40b0-ba14-2e0516a7a8ae.png)

**Comment by benjaminkohl** [comment]
I think I'm following along here. Having a custom relation field for Saved Filters seems like a cool idea. We'll have to ponder that a bit before deciding if that is something we want to add to the plugin.

**Comment by goraxan** [comment]
It is indeed. In my case I use Craft as a headless CMS, and I'm exposing search results (from a saved filter) in an API endpoint. 
The flexibility the plugin offers along with the possibility to extend it with your own metadata is awesome.

**Comment by benjaminkohl** [comment]
The Saved Filters relationship field is now available in [version 1.3.0](https://github.com/masugadesign/cp-filters-craft-cms/blob/master/CHANGELOG.md#130---2022-03-25).
